homeassistant/components/proximity/__init__.py

```diff
@@ -203,45 +209,48 @@ def check_proximity_state_change(self, entity, old_state, new_state):
             return
 
         # Collect distances to the zone for all devices.
-        distances_to_zone = {}
+        distances_to_zone: dict[str, float] = {}
         for device in self.proximity_devices:
             # Ignore devices in an ignored zone.
             device_state = self.hass.states.get(device)
-            if device_state.state in self.ignored_zones:
+            if not device_state or device_state.state in self.ignored_zones:
                 continue
 
             # Ignore devices if proximity cannot be calculated.
             if "latitude" not in device_state.attributes:
                 continue
 
             # Calculate the distance to the proximity zone.
-            dist_to_zone = distance(
+            distance_to_zone = distance(
```

Comment:
I would suggest to keep the old name. `distance_to_zone` can easily be confused with `distances_to_zone` in the same method.

Reply:
It is a local variable (within the loop) that was getting redifined on line 242 for the next loop.
I felt it was better to use two separate variables (one for the first loop, one for the second loop).
I've renamed to proximity, but you can suggest a new name.